### PR TITLE
Adding hosts module

### PR DESCRIPTION
**Pull request by jtyr**
##### ISSUE TYPE

New Module Pull Request.

##### COMPONENT NAME

`hosts`

##### ANSIBLE VERSION

For Ansible 2.3.

##### SUMMARY

This module adds the ability to add, remove and modify records in the `/etc/hosts` file.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @jtyr for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jtyr** [comment]
If somebody wants to test it, here is a recipe how to do that:
http://pastebin.com/ruBDuZpK

**Comment by jtyr** [comment]
Is this something you might like, @drybjed?

File: system/hosts.py [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+        tpm_fd, tmp_file = tempfile.mkstemp()
+
+        # Open the hosts file
+        try:
+            f = open(tmp_file, 'wb')
+        except IOError:
+            e = get_exception()
+            self.module.fail_json(
+                msg=(
+                    "Cannot open temporal hosts file %s for writing." %
```
Comment (drybjed) [inline system/hosts.py]: I would go with `temporary`, unless you go back in time... ;-)

**Review by drybjed — COMMENTED** [review]
Module looks good at first glance, I'll try to test it in a short while.

File: system/hosts.py (hunk 1 above)
Comment (jtyr) [reply system/hosts.py]: Good catch ;o) It's fixed.

**Pull request closed without merging**